rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala

```diff
@@ -126,9 +126,11 @@ abstract class Engine extends LazyLogging with JsonSupport {
   def input(json: String): Validated[ValidateError, Response] = {
     // flatten the event into one string per line as per Spark json collection spec
     mirroring.mirrorEvent(json.replace("\n", " ") + "\n")
-    Valid(Comment("Input processed by base Engine"))
+      .andThen(_ => Valid(Comment("Input processed by base Engine")))
   }
 
+  def inputMany: Seq[String] => Unit = _.foreach(input)
```

Comment:
Is this API extension important? It seems to only avoid one input per call. The call overhead should not be important, is it?

@andreyshikov

Reply:
It's not about call overhead, it's a sign that we can use a bulk insert that gives us a performance boost